src/Avalonia.Visuals/Animation/Transitions/BrushTransition.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using Avalonia.Animation.Animators;
+using Avalonia.Animation.Easings;
+using Avalonia.Media;
+
+#nullable enable
+
+namespace Avalonia.Animation
+{
+    /// <summary>
+    /// Transition class that handles <see cref="AvaloniaProperty"/> with <see cref="IBrush"/> type.
+    /// Only values of <see cref="ISolidColorBrush"/> will transition correctly at the moment.
+    /// </summary>
+    public class BrushTransition : Transition<IBrush?>
+    {
+        private static readonly ISolidColorBrushAnimator s_animator = new ISolidColorBrushAnimator();
+
+        public override IObservable<IBrush?> DoTransition(IObservable<double> progress, IBrush? oldValue, IBrush? newValue)
+        {
+            var oldSolidColorBrush = TryGetSolidColorBrush(oldValue);
+            var newSolidColorBrush = TryGetSolidColorBrush(newValue);
+
+            if (oldSolidColorBrush != null && newSolidColorBrush != null)
+            {
+                EnsureImmutable(ref oldSolidColorBrush);
+                EnsureImmutable(ref newSolidColorBrush);
+
+                return new AnimatorTransitionObservable<ISolidColorBrush, ISolidColorBrushAnimator>(
+                    s_animator, progress, Easing, oldSolidColorBrush, newSolidColorBrush);
+            }
+
+            return new IncompatibleTransitionObservable(progress, Easing, oldValue, newValue);
+        }
+
+        private static void EnsureImmutable(ref ISolidColorBrush brush)
+        {
+            brush = (ISolidColorBrush)brush.ToImmutable();
```

Comment:
Do we need this "ensure immutable" part at all? As I understand, you added it to avoid dependency properties usage from mutable brushes. But is it actually better than recreating new brush per each animation run?

Reply:
And I don't think we really need one line method here, even though it most likely will be inlined.

Reply:
I will remove it for now, it needs to allocate a new brush per transition run. Using a mutable variant will allocate eventually too since renderer works with immutable data anyway. Plus yeah accessing styled properties is slow but before benchmarking it is hard to tell.